### PR TITLE
Added Keystone restart service test

### PR DESCRIPTION
**Pull request by grebennikov**
Also modified PID filtering when restarting service to avoid
watching duplicate PIDs in containers

### CODE REVIEW

File: zaza/charm_tests/keystone/tests.py [hunk 1]
```diff
@@ -53,6 +53,29 @@ def test_pause_resume(self):
...
+        # Config file affected by juju set config change
+        conf_file = '/etc/keystone/keystone.conf'
+
+        # Make config change, check for service restarts
+        logging.debug('Setting debug mode keystone...')
+        self.restart_on_changed(
+            conf_file,
+            set_default,
+            set_alternate,
+            {'DEFAULT': {'debug': 'True'}},
```
Comment (fnordahl) [inline zaza/charm_tests/keystone/tests.py]: This should correspond to what you have in the `set_default` variable

File: zaza/model.py [hunk 2]
```diff
@@ -323,7 +323,8 @@ def get_unit_from_name(unit_name, model=None, model_name=None):
     :rtype: int
     :raises: ServiceNotRunning
     """
-    cmd = "stat -c %Y /proc/$(pidof -x {} | cut -f1 -d ' ')".format(service)
+    cmd = ("stat -c %Y /proc/$(pgrep {} --nslist pid"
```
Comment (fnordahl) [inline zaza/model.py]: This change is not directly connected to you adding a service restart test for the keystone charm. Please split out into separate PR or at least a separate commit.

Comment (ryan-beisner) [reply zaza/model.py]: @fnordahl please see https://github.com/openstack-charmers/zaza/pull/145

**Comment by grebennikov** [comment]
agreed that the change is not needed
